src/engine/resolvers.rs

```diff
@@ -80,6 +81,65 @@ where
         }
     }
 
+    /// Returns the resolver input deserialized into a structure of type T that
+    /// implements the serde `Deserialize` trait.
+    ///
+    /// # Errors
+    ///
+    /// Returns an [`Error]` variant [`InputNotFound`] if no input field was passed
+    /// to the query, [`TypeConversionFailed`] if unable to convert a warpgrapher Value
+    /// to a serde_json Value, and [`JsonDeserializationFaild`] if unable to parse the
+    /// input data into a struct of type T.
+    ///
+    /// [`Error`]: ../../error/enum.Error.html
+    /// [`InputNotFound`]: ../../error/enum.Error.html#variant.InputNotFound
+    /// [`TypeConversionFailed`]: ../../error/enum.Error.html#variant.TypeConversionFailed
+    /// [`JsonDeserializationFailed`]: ../../error/enum.Error.html#variant.JsonDeserializationFailed
+    pub fn parse_input<T: serde::de::DeserializeOwned>(&self) -> Result<T, Error> {
+        let value: Value = match self.args().get("input") {
+            None => {
+                return Err(Error::InputItemNotFound {
+                    name: "input".to_string(),
+                })
+            }
+            Some(v) => v,
+        };
+        let value2 = match serde_json::Value::try_from(value) {
+            Err(_) => {
+                return Err(Error::TypeConversionFailed {
+                    src: "warpgrapher::Value".to_string(),
+                    dst: "serde_json::Value".to_string(),
+                })
+            }
+            Ok(v) => v,
+        };
+        let parsed_input: T = match serde_json::from_value(value2) {
```

Comment:
This may get rolled into a chain of calls, per the comment above, but here in particular, you're unwrapping a successful result from an Ok, only to immediately re-wrap it in an Ok and return it.  It might be simpler as:

`serde_json::from_value(value2).map_err(|e| Error::JsonDeserializationFailed {source: e})`

Or... if we have the `From` traits set up correctly for a deserialization error, it would be simpler still:

`serde_json::from_value(value2).map_err(Error::from)`

Reply:
Fixed.